sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresNumericTests.java

```diff
@@ -920,6 +920,53 @@ SELECT x, sqrt(x)
                       4.2 | 2.049390153191920""");
     }
 
+    @Test
+    public void testExp() {
+        this.qs("""
+                --
+                -- Tests for EXP()
+                --
+                -- special cases
+                select exp(0.0);
+                        exp
+                --------------------
+                 1.0000000000000000
+                (1 row)
+                                
+                select exp(1.0);
+                        exp
+                --------------------
+                 2.7182818284590452
+                (1 row)
+                                
+                select exp(1.0::numeric(25, 10)); -- changed the precision and scale
+                        exp
+                --------------------
+                 2.7182818284590452
+                (1 row)
+                                
+                select exp(-32.999);
+                         exp
+                -----------------------
+                 4.663547361468238E-15
+                (1 row)
+
+                select exp(-123.456);
+                                                   exp
+                -------------------------------------------------------------------------
+                 0.000000000000000000000000000000000000000000000000000002419582541264601
+                (1 row)
+
+                -- big test
+                select exp(1234.5678);
+                 exp
+                -----
+                 146549072930959479983482138503979804217622199675223653966270157446954995433819741094410764947112047906012815540251009949604426069672532417736057033099274204598385314594846509975629046864798765888104789074984927709616261452461385220475510438783429612447831614003668421849727379202555580791042606170523016207262965336641214601082882495255771621327088265411334088968112458492660609809762865582162764292604697957813514621259353683899630997077707406305730694385703091201347848855199354307506425820147289848677003277208302716466011827836279231.9667
```

Comment:
how is this possible? This exceeds the precision of a Double. Most of these digits cannot be right

Reply:
Why is this printed with so many digits of precision? doubles should stop at 15

Reply:
I copied this from: https://github.com/postgres/postgres/blob/09eb633e1baa3b7cd7929f3cc77f9c46f63c20b1/src/test/regress/expected/numeric.out#L2976

I assume that internally it is ignored after 15 places. 
Its surprising that this doesn't give an error like the one above.

Reply:
you should check the generated rust

Reply:
The generated code is the same for both optimized and unoptimized versions: 
```rust
            Tup1::new(exp_d(cast_to_d_decimal(new_decimal("1234.5678", 8, 4).unwrap())))
```
`exp_d` returns infinity. 

The zset for the output is: 

```rust
        Tup1::new(F64::new(f64::from_bits(9218868437227405312u64))/*std::f64::INFINITY*/) => 1,
```

Which is also infinity.

Reply:
Then this is the test we have to run